packages/hash/frontend/src/pages/[account-slug]/[page-slug].page.tsx

```diff
@@ -39,50 +50,143 @@ import { PageIconButton } from "../../components/PageIconButton";
 import { PageLoadingState } from "../../components/PageLoadingState";
 import { CollabPositionProvider } from "../../contexts/CollabPositionContext";
 import {
+  GetAllLatestEntitiesQuery,
   GetPageInfoQuery,
   GetPageInfoQueryVariables,
 } from "../../graphql/apiTypes.gen";
+import { getAllLatestEntitiesQuery } from "../../graphql/queries/knowledge/entity.queries";
+import { apolloClient } from "../../lib/apollo-client";
+import { constructMinimalOrg, MinimalOrg } from "../../lib/org";
+import { constructMinimalUser, MinimalUser } from "../../lib/user";
 import { getLayoutWithSidebar, NextPageWithLayout } from "../../shared/layout";
 import { HEADER_HEIGHT } from "../../shared/layout/layout-with-header/page-header";
 import { useReadonlyMode } from "../../shared/readonly-mode";
-import { useRouteAccountInfo, useRoutePageInfo } from "../../shared/routing";
 import { Button } from "../../shared/ui/button";
 import {
   TOP_CONTEXT_BAR_HEIGHT,
   TopContextBar,
 } from "../shared/top-context-bar";
 
-// Apparently defining this is necessary in order to get server rendered props?
-export const getStaticPaths: GetStaticPaths<{ slug: string }> = () => ({
-  paths: [], // indicates that no page needs be created at build time
-  fallback: "blocking", // indicates the type of fallback
-});
-
 type PageProps = {
+  pageWorkspace: MinimalUser | MinimalOrg;
+  pageEntityId: EntityId;
   blocks: HashBlock[];
 };
 
+type PageParsedUrlQuery = {
+  "account-slug": string;
+  "page-slug": string;
+};
+
+export const tbdIsPageParsedUrlQuery = (
+  queryParams: NextParsedUrlQuery,
+): queryParams is PageParsedUrlQuery =>
+  typeof queryParams["account-slug"] === "string" &&
+  typeof queryParams["page-slug"] === "string";
+
+export const parsePageUrlQueryParams = (params: PageParsedUrlQuery) => {
+  const workspaceShortname = params["account-slug"].slice(1);
+
+  const pageEntityUuid = params["page-slug"];
+
+  return { workspaceShortname, pageEntityUuid };
+};
+
 /**
  * This is used to fetch the metadata associated with blocks that're preloaded
  * ahead of time so that the client doesn't need to
  *
  * @todo Include blocks present in the document in this
  */
-export const getStaticProps: GetStaticProps<PageProps> = async () => {
+export const getServerSideProps: GetServerSideProps<PageProps> = async ({
+  req,
+  params,
+}) => {
   const fetchedBlocks = await Promise.all(
     defaultBlockComponentIds.map((componentId) => fetchBlock(componentId)),
   );
 
+  if (!params || !tbdIsPageParsedUrlQuery(params)) {
+    throw new Error(
+      "Invalid page URL query params passed to `getServerSideProps`.",
+    );
+  }
+
+  const { workspaceShortname, pageEntityUuid } =
+    parsePageUrlQueryParams(params);
+
+  const cookieString = req?.headers.cookie;
```

Comment:
Should we call this `cookie` and let it be shorthanded below?

Reply:
The reason I went with `cookieString` was because `cookie` makes it sound like we're passing a singular cookie, whereas it's a string of all the concatenated cookies. I realise this will probably not make sense to other readers though, so have addressed in https://github.com/hashintel/hash/pull/1650/commits/d4eeef5aca3b86902d2cf4474b9cc01eaa8613e6